extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -346,4 +347,30 @@ public boolean isExtensionStoreInitialized() {
         return (storePath != null);
     }
 
+    public String updateExtensionStatus(final String extensionName, String currentUser, ExtensionStatus status) throws
+            FalconException {
+        validateStatusChange(extensionName, currentUser);
+        if (metaStore.getDetail(extensionName).getStatus().equals(status)) {
+            throw new ValidationException(extensionName + " is already in " + status.toString() + " state.");
```

Comment:
I recollect from the discussion we had, we decided asked to throw an error. I also stated the example of current status change of process/feed, where we do not throw an error.